src/Microsoft.ML.Onnx/SaveOnnxCommand.cs

```diff
@@ -232,13 +232,15 @@ private void Run(IChannel ch)
                 if (end.Schema.IsHidden(i))
                     continue;
 
-                var idataviewColumnName = end.Schema.GetColumnName(i);;
-                if (_outputsToDrop.Contains(idataviewColumnName) || _inputsToDrop.Contains(idataviewColumnName))
```

Comment:
> if (_outputsToDrop.Contains(idataviewColumnName) || _inputsToDrop.Contains(idataviewColumnName)) [](start = 15, length = 97)

Why did you remove inputsToDrop?

Reply:
We can't drop variables without considering the whole graph structure. For each non-hidden variable in output schema, it must either be computed somewhere or just an input. For the first case, we don't need to remove it. For the second case, we should connect that an input to that output via Identity.

---
In reply to: [222025561](https://github.com/dotnet/machinelearning/pull/952#discussion_r222025561) [](ancestors = 222025561)